go-controller/pkg/ovn/address_set_syncer/address_set_sync.go

```diff
@@ -0,0 +1,400 @@
+				// oldName is just namespace name
+				dbIDs = syncer.getNamespaceAddrSetDbIDs(oldName)
+			}
+		}
+	}
+	// dbIDs is set
+	return
+}
+
+func (syncer *addressSetsSyncer) getUpdateAddrSetOps(addrSetsInfo []*updateAddrSetInfo) (ops []libovsdb.Operation, err error) {
+	// one referencing object may contain multiple references that need to be updated
+	// these maps are used to track referenced that need to be replaced for every object type
+	aclsToUpdate := map[string]*nbdb.ACL{}
+	qosesToUpdate := map[string]*nbdb.QoS{}
+	lrpsToUpdate := map[string]*nbdb.LogicalRouterPolicy{}
+
+	for _, addrSetInfo := range addrSetsInfo {
+		if addrSetInfo.newAddrSet == nil {
+			// new address set wasn't built
+			if len(addrSetInfo.acls) == 0 && len(addrSetInfo.qoses) == 0 && len(addrSetInfo.lrps) == 0 {
+				// address set is stale and not referenced, clean up
+				ops, err = libovsdbops.DeleteAddressSetsOps(syncer.nbClient, ops, addrSetInfo.oldAddrSet)
+			} else {
+				syncer.ignoredAddressSets += 1
+			}
+			continue
+		}
+
+		oldName := addrSetInfo.oldAddrSet.ExternalIDs["name"]
+		// create updated address set
+		ops, err = libovsdbops.CreateOrUpdateAddressSetsOps(syncer.nbClient, ops, addrSetInfo.newAddrSet)
+		if err != nil {
+			return nil, fmt.Errorf("failed to get update address set ops for address set %s: %v", oldName, err)
+		}
+		// delete old address set
+		ops, err = libovsdbops.DeleteAddressSetsOps(syncer.nbClient, ops, addrSetInfo.oldAddrSet)
+		if err != nil {
+			return nil, fmt.Errorf("failed to get update address set ops for address set %s: %v", oldName, err)
+		}
+		oldHash := "$" + addrSetInfo.oldAddrSet.Name
+		newHash := "$" + addrSetInfo.newAddrSet.Name
+
+		for _, acl := range addrSetInfo.acls {
+			if _, ok := aclsToUpdate[acl.UUID]; !ok {
+				aclsToUpdate[acl.UUID] = acl
+			}
+			aclsToUpdate[acl.UUID].Match = strings.ReplaceAll(aclsToUpdate[acl.UUID].Match, oldHash, newHash)
+		}
+
+		for _, qos := range addrSetInfo.qoses {
+			if _, ok := qosesToUpdate[qos.UUID]; !ok {
+				qosesToUpdate[qos.UUID] = qos
+			}
+			qosesToUpdate[qos.UUID].Match = strings.ReplaceAll(qosesToUpdate[qos.UUID].Match, oldHash, newHash)
+		}
+
+		for _, lrp := range addrSetInfo.lrps {
+			if _, ok := lrpsToUpdate[lrp.UUID]; !ok {
+				lrpsToUpdate[lrp.UUID] = lrp
+			}
+			lrpsToUpdate[lrp.UUID].Match = strings.ReplaceAll(lrpsToUpdate[lrp.UUID].Match, oldHash, newHash)
+		}
+	}
+
+	for _, acl := range aclsToUpdate {
+		ops, err = libovsdbops.CreateOrUpdateACLsOps(syncer.nbClient, ops, acl)
+		if err != nil {
+			return nil, fmt.Errorf("failed to get update acl ops: %v", err)
+		}
+	}
+	for _, qos := range qosesToUpdate {
+		ops, err = libovsdbops.CreateOrUpdateQoSesOps(syncer.nbClient, ops, qos)
+		if err != nil {
+			return nil, fmt.Errorf("failed to get update qos ops: %v", err)
+		}
+	}
+	for _, lrp := range lrpsToUpdate {
+		ops, err = libovsdbops.UpdateLogicalRouterPoliciesOps(syncer.nbClient, ops, lrp)
```

Comment:
So I guess it was a conscious decision not to handle the ErrNotFound error here. I guess it makes sense since nobody should be deleting those and if it happens we would just crash-loop. I actually like this behavior best.

I still think we would be better to behave in an equivalent way with ACLs and QoS. Rather, on the same circumstance,   if someone had deleted them, we would silently move on with a noop. This shouldn't happen but if it did happen it could probably mean that the AS itself shouldn't exist either and we would be missing the fact.

Reply:
yeah i agree it would be better, but that would mean that some other ops e.g. on sync for acls should be updated too. So it would be nice to handle that in another PR I think

Reply:
The consequences might not be as severe for those other ops because they are standalone ops, they don't bundle other ops in the transaction as we do here updating the address sets.
I don't mind this to be done in another PR but I would be mindful on the release itself, in the case there is a need to handle this concern.